controllers/operatorpolicy_controller.go

```diff
@@ -114,12 +115,42 @@ func (r *OperatorPolicyReconciler) SetupWithManager(mgr ctrl.Manager, depEvents
 		For(
 			&policyv1beta1.OperatorPolicy{},
 			builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(
+			&policyv1beta1.OperatorPolicy{},
+			handler.EnqueueRequestsFromMapFunc(overlapMapper)).
 		WatchesRawSource(
 			depEvents,
 			&handler.EnqueueRequestForObject{}).
 		Complete(r)
 }
 
+func overlapMapper(_ context.Context, obj client.Object) []reconcile.Request {
+	//nolint:forcetypeassert
+	pol := obj.(*policyv1beta1.OperatorPolicy)
+
+	var result []reconcile.Request
+
+	for _, overlap := range pol.Status.OverlappingPolicies {
```

Comment:
Would it be cleaner to have `OverlappingPolicies` be a struct with the name and namespace field separated?

Reply:
Being able to use `slices.Sort`, `slices.Equal` and `strings.Join` on them as they are right now is kind of nice. It wouldn't be too difficult to get those same kinds of things on a new struct, but I don't think we would gain much overall.